Fix for issue 984 and 992

AdamBJ opened the pull request:
## PLEASE READ THIS BEFORE MERGING THE PULL REQUEST
This PR populates 15 previously empty expected output files used in the JUnit template tests. These files are associated primarily with state machines. Before merging this PR, someone needs to verify that the output I've added is correct given the existing input `.ump` files. I've done my best to do this, but I'm not that familiar with state machines.

For example, someone needs to look at `stateMachineWithStringComparisonGuard.ump` and verify that the output added to `stateMachineWithStringComparisonGuard.php.txt` is the output one would expect if Umple was run with `stateMachineWithStringComparisonGuard.ump`. There are around 15 such verification steps required. I think this might be a good task for @jblang94, it might be best to create a new issue to address this task. Please see [this commit](https://github.com/umple/umple/commit/46ec5e94c8aa5ed015f9c99ccfc3a5301f29e6ce) for a complete list of the expected output files that were populated.

## Description

This PR adds a check to `assertFileContent` to make sure that the expected output file in not empty. Adding this check revealed 15 expected output files that were empty. I have run Umple on the input files associated with these empty expected output files and populated them with the results. Verification is required before this PR can be merged with master.

This PR also fixes [issue 992](https://github.com/umple/umple/issues/992) (see [this commit](https://github.com/umple/umple/commit/c1b86996652d57684b02df560b7efb88beb0438e)).

## Tests

This PR populated all the empty expected output files uncovered by the issue 984 fix, and deleted one duplicate test. It also modifies an existing test case and adds more test cases to ensure that the fix introduced to address issue 992 is correct.

`Closes #992, #984`

a3994288 committed:
ignore the operation that starts with exlamation

a3994288 committed:
added warning when exclusion method is not found

a3994288 committed:
fixed bug

a3994288 committed:
warning 1014 should not be raised in some cases

a3994288 committed:
added tests

AdamBJ committed:
- Fixed bug in SampleFileWriter_Code.ump that caused empty expected output files to go unreported. This bug has led to approximately 15 empty expected outputs files being included in the test suites.
- Ran Umple for each of the empty test files and recorded the output in the appropriate empty expected output file (this have not been carefully checked, I'll be looking for help from more experienced contributor on this)
- Removed one test file, ExampleFile.ump. This test case was exactly the same as the test case directly below it in AlloyTemplateTest.java
- A static variable controlling the numbering of auto transitions was being carried over from test to test and producing variable outputs in StateMachineTest.java. Added code in StateMachineTest.java to clear the state of the Event object after each test.

AdamBJ committed:
- Fixed bug in SampleFileWriter_Code.ump that caused empty expected output files to go unreported. This bug has led to approximately 15 empty expected outputs files being included in the test suites.
- Ran Umple for each of the empty test files and recorded the output in the appropriate empty expected output file (this have not been carefully checked, I'll be looking for help from more experienced contributor on this)
- Removed one test file, ExampleFile.ump. This test case was exactly the same as the test case directly below it in AlloyTemplateTest.java
- A static variable controlling the numbering of auto transitions was being carried over from test to test and producing variable outputs in StateMachineTest.java. Added code in StateMachineTest.java to clear the state of the Event object after each test.

AdamBJ committed:
-Fixed recursive inheritance problem for interfaces (issue 992).
-Added tests to verify correct 992 fix

AdamBJ committed:
-Fixed recursive inheritance problem for interfaces (issue 992).
-Added tests to verify correct 992 fix

AdamBJ committed:
Added test to ensure that a class that implements a "recursive" interface doesn't explicitly include the interface's methods in its generated code.

AdamBJ committed:
Added test to ensure that a class that implements a "recursive" interface doesn't explicitly include the interface's methods in its generated code.

AdamBJ committed:
Removed .txt file associated with alloy/ExampleFile.ump (forgot to do this when removing the original .ump file).

AdamBJ committed:
Fixing indentation issue in UmpleInternalParser_CodeClass.ump, and attempting to fix travis build failure by seperating tests in JavaInterfaceTemplateTest.java.

AdamBJ committed:
Fixing indentation issue in UmpleInternalParser_CodeClass.ump, and attempting to fix travis build failure by seperating tests in JavaInterfaceTemplateTest.java.

AdamBJ commented on cruise.umple/src/UmpleInternalParser_CodeClass.ump:
```diff
@@ -1602,20 +1602,20 @@ private Boolean checkIsDistributed(UmpleInterface uInterface)
+		{
+		  for (Method aMethod : pi.getMethods())
+		  {
+			boolean shouldAddMethod = verifyIfMethodIsConstructorOrGetSet(uClass, aMethod, true);
```
This looks fine in my local editor (Notepad++). Working on a fix.

Nava2 replied:
Check if your editor is set to tabs vs. spaces, you should use spaces and 2 per tab! 👍

AdamBJ committed:
Format and travis build fix attempt #2

AdamBJ committed:
Format and travis build fix attempt #2

AdamBJ committed:
Added missing expected output file that was causing the Travis failure

AdamBJ committed:
Added missing expected output file that was causing the Travis failure

AdamBJ committed:
Fixes interface hierarchy code so that it properly traverses the entire interface hierarchy. Adding additional tests to verify the fix.

AdamBJ committed:
Fixes interface hierarchy code so that it properly traverses the entire interface hierarchy. Adding additional tests to verify the fix.

Jaorji committed:
Fix#482 issue for Umple

vahdat-ab committed:
Merge pull request #994 from umple/Fix-#483

Fix #483 issue for Umple

opeyemiAdesina committed:
fixing deleted test case for issue #993

opeyemiAdesina committed:
Merge branch 'master' into FormalUmple-oades013

opeyemiAdesina committed:
Merge pull request #996 from umple/FormalUmple-oades013

fixing deleted test case for issue #993

jblang94 committed:
Add entry to en.error, and test cases

jblang94 committed:
Add detection for parallel state machines that have the same name in the same nested state machine
* Modify test case structure
* Add separate error for duplicate parallel state machines in the same region

jblang94 committed:
Add addNewSm to conditional check for issue 778

jblang94 committed:
Add issue 778 detection for parallel regions
* Duplicate name is now provided in both error messages
* Added checkParallelRegionsForParallelSmsWithSameName method

jblang94 committed:
Add additional test cases

jblang94 committed:
Add User Manual entries for E073 and E074
* Add user manual pages and examples for E073 and E074
* Add additional test case which is also used as a user manual example
* Modify E074 error message
* Update links in en.error for E073 and E074
* Add missing link for W072 (from my past fix for issue #925)

AdamBJ committed:
Added code to delete Object.php in php/PhpClassTemplateTest.java's tearDown method so that Object.php doesn't clutter the ../implementation/php/ folder.

vahdat-ab committed:
Merge pull request #998 from AdamBJ/Object.php_fix

Remove Object.php

AdamBJ committed:
Removed check for input parameter name

vahdat-ab committed:
Merge pull request #999 from AdamBJ/remove_arg_name_check

Removed check for input parameter name

amidzak committed:
new distributed pattern and several bug fixes (#1000)

* fixed a bug in object factory related to package import

* fixed a bug in scanning the generated code

* fixed object factory's problem with user defined constructors

* rmi initiation moved to object factory

Another distributed pattern added

* full build

Web service RPC  added(compiles but not running)
some modifications to RPC
* fixing construcor name for distributed classes with name change
* fix distributed test to delete the files

jblang94 committed:
Remove test case and code that generates E074

jblang94 committed:
Remove all references to E074
* Remove all test files
* Remove all user manual examples
* Remove E074 user manual entry
* Remove E074 from en.error

jblang94 committed:
Update E073 user manual entry and en.error entry
* Rename the error to "Duplicate Parallel State Machine Name"
* Rename the user manual example files
* Adjust en.error and the manual entry to use composite state in description

jblang94 committed:
Update comment in UmpleInternalParser_CodeStateMachine.ump

vahdat-ab committed:
Merge pull request #977 from umple/fix#682

Fix#682 - added warning for exclusion operation

vahdat-ab committed:
Merge pull request #997 from jblang94/resolve_issue_778

Resolve issue 778

romantic668 committed:
Update README.md

Hi, I am doing a technical writing project. I just noticed a few grammar issues. Other than that, it is perfect.

vahdat-ab committed:
Merge pull request #1003 from romantic668/master

Update README.md

jblang94 committed:
Initial test case parallelSm_sameNameDiffStates passes

jblang94 committed:
Modify fix and update test cases
* Add comparison between the parent state's name and its state machine name
* Add additional Java code generation tests
* Add PHP code generation tests

jblang94 committed:
Update issue 1001 test cases
* Add 2 test cases that include entry and exit actions in the parallel state machines
* Fix test cases to use @UMPLE_VERSION@

jblang94 committed:
Update prepare_postpare_concurrentStateMachine in JavaGeneratorTest

jblang94 committed:
Update NuSMVTemplateTest test cases

jblang94 committed:
Update StateMachines test in UmpleSelfGeneratorTest

jblang94 committed:
Update StateMachineTemplateTest test cases

jblang94 committed:
Update testbed tests that had parallel state machines

vahdat-ab committed:
Merge pull request #1004 from jblang94/resolve_issue_1001

Resolve issue 1001

AdamBJ committed:
Removed duplicate JUnit test. See https://github.com/umple/umple/pull/993#issuecomment-279792939

AdamBJ committed:
Merge branch '984' of https://github.com/AdamBJ/umple into 984

# Conflicts:
#	cruise.umple/src/UmpleInternalParser_CodeClass.ump
